### PR TITLE
[ENG-3913] Add new resource

### PR DESCRIPTION
**Pull request by adlius**
-   Ticket: []
-   Feature flag: n/a

### CODE REVIEW

File: lib/osf-components/addon/components/resources-list/edit-resource/component.ts [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            presence: true,
+            translationArgs: { description: this.intl.t('osf-components.resources-list.edit_resource.output_type') },
+        })],
+        description: [validatePresence({
+            type: 'blank',
+            presence: true,
+            translationArgs: { description: this.intl.t('osf-components.resources-list.edit_resource.description') },
+        })],
+    };
+
```
Comment (brianjgeiger) [inline lib/osf-components/addon/components/resources-list/edit-resource/component.ts:52]: Shouldn't there be a `title` field as well?

File: lib/osf-components/addon/components/resources-list/edit-resource/template.hbs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+                    @label={{t 'osf-components.resources-list.edit_resource.output_type'}}
+                    @options={{this.availableTypes}}
+                    @searchEnabled={{false}}
+                    @renderInPlace={{true}}
+                    as |type|
+                >
+                    {{t (concat 'osf-components.resources-list.edit_resource.' type)}}
+                </form.select>
+                <form.textarea
+                    @isRequired={{true}}
```
Comment (brianjgeiger) [inline lib/osf-components/addon/components/resources-list/edit-resource/template.hbs:37]: Is description actually required?

File: lib/osf-components/addon/components/resources-list/edit-resource/template.hbs [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+                    @valuePath={{'description'}}
+                    @label={{t 'osf-components.resources-list.edit_resource.description'}}
+                />
+            </FormControls>
+        {{/unless}}
+        {{#if this.shouldShowPreview}}
+            <p>
+                {{t 'osf-components.resources-list.edit_resource.check_your_doi'}}
+            </p>
+            <ResourcesList::ResourceCard @resource={{this.resource}} @eidtable={{false}} />
```
Comment (brianjgeiger) [inline lib/osf-components/addon/components/resources-list/edit-resource/template.hbs:47]:
```suggestion
            <ResourcesList::ResourceCard @resource={{this.resource}} @editable={{false}} />
```

**Review by brianjgeiger — CHANGES_REQUESTED** [review]
Quick first pass; a couple of questions and one typo. Looking good overall.

File: lib/osf-components/addon/components/resources-list/edit-resource/component.ts (hunk 1 above)
Comment (adlius) [reply lib/osf-components/addon/components/resources-list/edit-resource/component.ts:52]: There isn't in the mockup though. The `resource-card` doesn't show title as well.

Comment (brianjgeiger) [reply lib/osf-components/addon/components/resources-list/edit-resource/component.ts:52]: Hmm. Might be something to discuss at scrum. I think both Jon and I thought there would be, but yeah, maybe not.

File: lib/osf-components/addon/components/resources-list/edit-resource/template.hbs [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+                    {{t (concat 'osf-components.resources-list.edit_resource.' type)}}
+                </form.select>
+                <form.textarea
+                    @isRequired={{true}}
+                    @valuePath={{'description'}}
+                    @label={{t 'osf-components.resources-list.edit_resource.description'}}
+                />
+            </FormControls>
+        {{/unless}}
+        {{#if this.shouldShowPreview}}
```
Comment (futa-ikeda) [inline lib/osf-components/addon/components/resources-list/edit-resource/template.hbs:43]: Can we move the above `{{#unless this.shouldShowPreview}}`  to the `{{else}}` block of this condition?

File: lib/osf-components/addon/components/resources-list/template.hbs [hunk 5]
```diff
@@ -14,7 +30,7 @@
         local-class='ResourceList'
 >
     <:item as |resource|>
-        <ResourcesList::ResourceCard @resource={{resource}} />
+        <ResourcesList::ResourceCard @resource={{resource}} @editable={{true}} />
```
Comment (futa-ikeda) [inline lib/osf-components/addon/components/resources-list/template.hbs:33]: We'll need to check if `@registration.userHasWritePermission` to see if it should be editable.

File: translations/en-us.yml [hunk 6]
```diff
@@ -1716,6 +1716,15 @@ osf-components:
         data: Data
         materials: Materials
         no-resources: 'No resources found.'
+        edit_resource:
+            preview: 'Preview'
+            heading: 'Add new output'
+            doi: 'DOI'
+            output_type: 'Output type'
```
Comment (futa-ikeda) [inline translations/en-us.yml:1723]:
```suggestion
            heading: 'Add new resource'
            doi: 'DOI'
            output_type: 'Resource type'
```

**Review by futa-ikeda — CHANGES_REQUESTED** [review]
Looks great! Just a few minor things I saw

File: lib/osf-components/addon/components/resources-list/template.hbs [hunk 7]
```diff
@@ -1,8 +1,24 @@
...
+            @reload={{this.reload}}
+            as |modal|
+        >
+            <div>
+                <Button
+                    local-class='TriggerButton'
+                    {{on 'click' modal.open}}
+                >
+                    <FaIcon @icon='plus' @size='2x' @fixedWidth={{true}}/>
+                </Button>
```
Comment (brianjgeiger) [inline lib/osf-components/addon/components/resources-list/template.hbs:19]: This should only be visible if the user has Write permission.

File: lib/osf-components/addon/components/resources-list/edit-resource/component.ts [hunk 8]
```diff
@@ -0,0 +1,104 @@
+import { action } from '@ember/object';
+import { inject as service } from '@ember/service';
+// import { waitFor } from '@ember/test-waiters';
+import Component from '@glimmer/component';
+import { ValidationObject } from 'ember-changeset-validations';
+import { validatePresence } from 'ember-changeset-validations/validators';
+import { task } from 'ember-concurrency';
+// import { taskFor } from 'ember-concurrency-ts';
```
Comment (brianjgeiger) [inline lib/osf-components/addon/components/resources-list/edit-resource/component.ts:8]: These comments should probably be removed.

File: lib/osf-components/addon/components/resources-list/edit-resource/component.ts [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+        this.shouldShowPreview = false;
+    }
+
+    @task
+    async finalize() {
+        if (this.resource) {
+            this.resource.finalized = true;
+            await this.resource.save();
+        }
+    }
```
Comment (brianjgeiger) [inline lib/osf-components/addon/components/resources-list/edit-resource/component.ts:103]: I think this is supposed to show toast messages in the success and failure cases.

**Review by brianjgeiger — CHANGES_REQUESTED** [review]
Overall looking great. A couple bits of missing functionality, and some commented out code.

**Review by brianjgeiger — APPROVED** [review]

File: lib/osf-components/addon/components/resources-list/template.hbs [hunk 10]
```diff
@@ -1,8 +1,26 @@
...
-    <div data-test-add-resource-section>
-        {{t 'osf-components.resources-list.add_instructions'}}
+    <div
+        local-class='Flexbox'
+        data-test-add-resource-section
+    >
+        <span>{{t 'osf-components.resources-list.add_instructions'}}</span>
         {{!-- help icon --}}
-        {{!-- add button --}}
+        {{#if @registration.userHasWritePermission}}
```
Comment (futa-ikeda) [inline lib/osf-components/addon/components/resources-list/template.hbs:8]: This condition is redundant

File: translations/en-us.yml [hunk 11]
```diff
@@ -1715,6 +1715,17 @@ osf-components:
...
         supplements: 'Supplements'
         no-resources: 'No resources found.'
+        edit_resource:
+            preview: 'Preview'
+            heading: 'Add new resource'
+            doi: 'DOI'
+            output_type: 'resource type'
+            description: 'Description'
+            data: 'Data'
+            materials: 'Materials'
```
Comment (futa-ikeda) [inline translations/en-us.yml:1725]: We will need to add Analytic Code, Papers, and Supplements here as well

**Review by futa-ikeda — CHANGES_REQUESTED** [review]
A couple very minor things. Rest looks good!

File: lib/osf-components/addon/components/resources-list/template.hbs (hunk 7 above)
Comment (brianjgeiger) [reply lib/osf-components/addon/components/resources-list/template.hbs:19]: You reverted this back with your last push. Did you mean to do that?

Comment (brianjgeiger) [reply lib/osf-components/addon/components/resources-list/template.hbs:19]: Nvm. I see it now.

**Review by futa-ikeda — APPROVED** [review]

**Pull request merged**